common/zt_logSanitizer_test.go

```diff
@@ -0,0 +1,68 @@
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package common
+
+import (
+	chk "gopkg.in/check.v1"
+)
+
+type logSanitizerSuite struct{}
+
+var _ = chk.Suite(&logSanitizerSuite{})
+
+func (s *logSanitizerSuite) TestLogSanitizer(c *chk.C) {
+
+	cases := []struct {
+		raw               string
+		expectedSanitized string
+	}{
+		{"string with no secrets", "string with no secrets"},
+
+		// DON'T redact if its not part of a name-value pair
+		{"This is the sig that I have and x=y", "This is the sig that I have and x=y"},
+
+		{"http://foo?sig=somevalue&x=y", "http://foo?sig=-REDACTED-&x=y"},                                         // remainder of query string is preserved
+		{"http://foo?x=y&" + SigAzure + "=somevalue", "http://foo?x=y&sig=-REDACTED-"},                            // basic case
+		{"http://foo?x=y&sig=somevalue\r\nBlah", "http://foo?x=y&sig=-REDACTED-\r\nBlah"},                         // newline after, case preserved in other text
+		{"blah\r\nhttp://foo?x=y&sig=somevalue blah", "blah\r\nhttp://foo?x=y&sig=-REDACTED- blah"},               // newline before and something else after
+		{"http://foo?a=b&" + SigXAmzForAws + "=somevalue&x=y", "http://foo?a=b&x-amz-signature=-REDACTED-&x=y"},   // AWS (using our official constant for the key
+		{"http://foo?a=b&X-Amz-siGnature=somevalue&x=y", "http://foo?a=b&X-Amz-siGnature=-REDACTED-&x=y"},         // weird caps
+		{"http://foo?a=b&someother-signature=somevalue&x=y", "http://foo?a=b&someother-signature=-REDACTED-&x=y"}, // unexpected sig type
+		{"http://foo?x=y&my-token=somevalue", "http://foo?x=y&my-token=-REDACTED-"},                               // name ending in "token"
+		{"Foo=x;Signature=bar", "Foo=x;Signature=-REDACTED-"},                                                     // not in a query string
+		{"Signature = bar;Foo = x", "Signature = -REDACTED-;Foo = x"},                                             // not in a query string, with spaces
+		{"Foo=x;Signature=bar", "Foo=x;Signature=-REDACTED-"},                                                     // not in a query string
+		{"Foo : x, Signature : bar, Other: z", "Foo : x, Signature : -REDACTED-, Other: z"},                       // not in a query string, with commas and spaces
+
+		// two replacements in same string
+		{"http://foo?sig=somevalue and http://bar?sig=othervalue BlahBlah", "http://foo?sig=-REDACTED- and http://bar?sig=-REDACTED- BlahBlah"},
+
+		// word "sig" inside the signature
+		{"http://foo?sig=sigvalue BlahBlah", "http://foo?sig=-REDACTED- BlahBlah"},
```

Comment:
consider adding cases where `sig` or `signature` is not part of the query and shouldn't be redacted.

Reply:
Done